[Jeff] iP

JeffinsonDarmawan opened the pull request.

JeffinsonDarmawan committed:
modified Duke.java

JeffinsonDarmawan committed:
changed chat bot name to Spock

JeffinsonDarmawan committed:
Level 1

JeffinsonDarmawan committed:
Level 2

JeffinsonDarmawan committed:
Level 2

JeffinsonDarmawan committed:
Level 3

with some future TODO for improvements

wjunjie01 commented on src/main/java/Spock.java:
```diff
@@ -0,0 +1,62 @@
+
+public class Spock {
+    public static void main(String[] args) {
+        System.out.println("____________________________________________________________");
```
Can make the horizontal line into a final constant (_______)

awesomesjh replied:
Perhaps replace "____________________________________________________________" with a constant to make the code more readable?
```suggestion
        System.out.println(LINE);
```

wjunjie01 commented on src/main/java/Spock.java:
```diff
@@ -0,0 +1,62 @@
+                reply = new Task(input, false, i);
+                list.add(reply);
+                System.out.println("added: " + reply.description);
+                i++;
```
Can replace the variable 'i' with a clearer variable like task count to keep track of number of task

wjunjie01 commented on src/main/java/Spock.java:
```diff
@@ -0,0 +1,62 @@
+        String input = myObj.nextLine();
+        List<Task> list = new ArrayList<>();
+        Task reply = new Task(input, false, 1);
+
+
+
```
Unnecessary spaces throughout code

wjunjie01 reviewed (commented):
Overall looks okay, can afford to extract methods and refactor code to make it easier to read.

awesomesjh commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,22 @@
+    protected boolean isDone;
+    protected Integer index;
+
+    public Task(String description, boolean b, int i) {
+        this.description = description;
+        this.isDone = b;
+        this.index = i;
+    }
```
Perhaps more intuitive variable names would be isDone and index?
```suggestion
    public Task(String description, boolean isDone, int index) {
        this.description = description;
        this.isDone = isDone;
        this.index = index;
    }
```

Fureimi replied:
I agree with this comment, will be easier for readability as well as making changes to the code in the future

awesomesjh commented on src/main/java/Spock.java:
```diff
@@ -0,0 +1,62 @@
+
+            if (input.equals("list")) {
+                System.out.println("Here are the tasks in your list:");
+                for (int k = 0; k < i-1; k++) {
```
Perhaps use j as an iterator? Since iterator naming precedence is i, j, k.
```suggestion
                for (int j = 0; j < i - 1; j++) {
```

awesomesjh commented on src/main/java/Spock.java:
```diff
@@ -0,0 +1,62 @@
+                    System.out.println(list.get(k).index + ".[" + list.get(k).getStatusIcon() + "] " + list.get(k).description);
+                }
+            }
+            else if (input.startsWith("mark")) { //TODO: disallow marking a task that is already marked and task_number outside of range
```
I believe else should start on the same line as }. I noticed the same issue in several other places too.
```suggestion
            } else if (input.startsWith("mark")) {
```

awesomesjh commented on src/main/java/Spock.java:
```diff
@@ -0,0 +1,62 @@
+                }
+            }
+            else if (input.startsWith("mark")) { //TODO: disallow marking a task that is already marked and task_number outside of range
+                int task_number = Integer.parseInt(input.replace("mark ", "")) - 1;
```
I think variable names should be in camelCase.
```suggestion
                int taskNumber = Integer.parseInt(input.replace("mark ", "")) - 1;
```

Fureimi replied:
Yes, I believe variable names should be in camelCase

awesomesjh reviewed (commented):
More can be done to improve readability of code, consider adding more classes and methods. Some violations of Java coding standard, consider rectifying them to ensure code consistency.

Fureimi commented on src/main/java/Spock.java:
```diff
@@ -1,11 +1,23 @@
+            System.out.println("____________________________________________________________");
+        }
+
+
```
Is there a reason for 2 lines spacing here?

Fureimi commented on src/main/java/Spock.java:
```diff
@@ -1,11 +1,23 @@
+        System.out.println("____________________________________________________________");
+        String input = myObj.nextLine();
+
+        for(String reply = input; !reply.equals("bye"); reply = myObj.nextLine()) {
```
I like how you the input in a for loop, makes your code neater.

Fureimi reviewed (commented):
Overall looks good

Fureimi commented on src/main/java/Spock.java:
```diff
@@ -11,26 +12,42 @@ public static void main(String[] args) {
+                    System.out.println(list.get(k).index + ".[" + list.get(k).getStatusIcon() + "] " + list.get(k).description);
                 }
             }
+            else if (input.startsWith("mark")) { //TODO: disallow marking a task that is already marked and task_number outside of range
```
May be better to remove such comments once the task is done

Fureimi reviewed (commented):
Overall good, some minor naming convention mistakes

JeffinsonDarmawan committed:
Level 3

with some future TODO for improvements

JeffinsonDarmawan committed:
Level 4

Added subclasses Todo, Deadline and Event

JeffinsonDarmawan committed:
Level 5

Handling Errors using custom exception

aditichadha1310 commented on src/main/java/Chandler.java:
```diff
@@ -0,0 +1,113 @@
+    public static final String LINE_DIVIDER = "____________________________________________________________";
+    public static final String OUTPUT_INDENTATION = "    ";
+
+    public static void main(String[] args) {
```
Avoid long methods. Take corrective action when the method exceeds 30 Lines of Code.

aditichadha1310 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    public String toString() {
+        return "[" + this.getStatusIcon() + "] " + this.description;
```
Avoid the usage of Magic Literals.

aditichadha1310 commented on src/main/java/Chandler.java:
```diff
@@ -0,0 +1,113 @@
+                        String by = input.substring(indexBy + 4);
+                        Deadline taskDeadline = new Deadline(description, by, "D");
+                        list.add(taskDeadline);
+                        System.out.println(
```
Good work with keeping each line of code within the character limits prescribed for each line.

aditichadha1310 commented on src/main/java/Chandler.java:
```diff
@@ -0,0 +1,113 @@
+                String[] words = input.split("\\s+");
+                String inputCommand = words[0];
+
+                if (words.length <= 1 && !inputCommand.equals("list") && !inputCommand.equals("bye") && !inputCommand.equals("mark") && !inputCommand.equals("unmark")) {
```
This is a fairly complex code statement. To improvise the readability for the user, you might want to break this down to simpler statements and use boolean variables to track each sub-expression.

aditichadha1310 reviewed (commented):
Overall, good job!

JeffinsonDarmawan committed:
Level 6

Added a feature on deleting tasks

JeffinsonDarmawan committed:
1. Level-7: Save, added the ability to save the list to the hard disk
 2. Rectified the overuse and inappropriate use of static variables and methods
 3. Code refactoring by creating more classes and shortening existing methods to improve readibility and consistency

JeffinsonDarmawan committed:
Committing some leftover changes

JeffinsonDarmawan committed:
Level-5 Exception Handling

JeffinsonDarmawan committed:
Leftover changes

JeffinsonDarmawan committed:
Exception Handling

JeffinsonDarmawan committed:
Level-9 find

JeffinsonDarmawan committed:
jar related commit

JeffinsonDarmawan committed:
Merge branch 'master' into branch-level-9

JeffinsonDarmawan committed:
debugged minor errors

JeffinsonDarmawan committed:
Merge branch 'master' into branch-level-9

# Conflicts:
#	src/main/java/TaskList.java

JeffinsonDarmawan committed:
Merge pull request #1 from JeffinsonDarmawan/branch-level-9

Branch level 9

JeffinsonDarmawan committed:
A-MoreOOP added Storage, Ui and Parser as well as modified the code of the main class

JeffinsonDarmawan committed:
Made several edits to ensure codes comply with the appropriate coding standard to enhance code quality.
Added more exception handling clauses.

JeffinsonDarmawan committed:
Added JavaDoc comments to all methods

JeffinsonDarmawan committed:
Add JavaDoc comments for classes

JeffinsonDarmawan committed:
Added JavaDoc comments

JeffinsonDarmawan committed:
Merge branch 'branch-A-JavaDoc'

JeffinsonDarmawan committed:
Modified README file

JeffinsonDarmawan committed:
Edit User Guide

JeffinsonDarmawan committed:
Update file name

JeffinsonDarmawan committed:
Update _config.yml

JeffinsonDarmawan committed:
Update _config.yml

JeffinsonDarmawan committed:
Update _config.yml

JeffinsonDarmawan committed:
Edit ReadMe file

JeffinsonDarmawan committed:
Merge branch 'master' of https://github.com/JeffinsonDarmawan/ip

JeffinsonDarmawan committed:
META-INF

JeffinsonDarmawan committed:
Update README.md
